### ISSUE TITLE
MPS: Progressive AR inference degradation on sequential synthesize() calls without torch.mps.empty_cache()

### ISSUE THREAD

**Issue by devinmlowe** [devinmlowe/chatterstream-tts#2]
## Summary

When running sequential `StreamingTTS.synthesize()` calls on MPS (Apple Silicon), autoregressive inference time degrades progressively — growing from ~1s to ~40s+ by the 5th-6th call. The degradation is caused by MPS intermediate tensor accumulation when `torch.mps.empty_cache()` is not called between synthesis invocations.

## Environment

- **Device**: Apple M4 (Mac Mini)
- **Backend**: MPS (Metal Performance Shaders)
- **Python**: 3.11
- **PyTorch**: Latest (MPS backend)
- **ChatterStream**: Latest from pip

## Reproduction

```python
import asyncio
import time
from chatterstream import StreamingTTS

async def main():
    tts = StreamingTTS(device="mps", watermark=False)
    tts.load()

    sentences = [
        "Hello, how are you?",
        "The quick brown fox jumps over the lazy dog.",
        "I can help you with a wide variety of tasks.",
        "Machine learning models have become increasingly capable.",
        "The weather forecast calls for partly cloudy skies.",
        "Let me tell you about the history of computing.",
    ]

    for i, text in enumerate(sentences):
        start = time.monotonic()
        async for chunk in tts.synthesize(text, voice="builtin"):
            pass
        elapsed = time.monotonic() - start
        print(f"Call {i+1}: {elapsed:.1f}s — {text[:50]}")

asyncio.run(main())
```

## Observed Behavior

Sequential calls show AR inference time growing exponentially:

| Call | Text | AR Inference Time | Total Synthesis Time |
|------|------|-------------------|---------------------|
| 1 | "Hello, how are you?" | ~973ms | ~2.6s |
| 2 | "The quick brown fox..." | ~2,556ms | ~3.8s |
| 3 | "I can help you with..." | ~9,760ms | ~12.0s |
| 4 | "Machine learning models..." | ~35,115ms | ~40.0s |
| 5 | "The weather forecast..." | ~40,165ms | ~45.0s |

The degradation is specifically in the autoregressive token generation step (T3 model). Vocoding time remains relatively stable. By the 5th call, AR inference is **~40x slower** than the first call.

## Expected Behavior

Sequential synthesis calls should maintain roughly consistent AR inference times (within ~2x of the first post-warmup call).

## Root Cause Analysis

The MPS backend accumulates intermediate tensors across synthesis calls. Without explicit cache clearing, the Metal allocator becomes fragmented and progressively slower. Key observations:

1. **MPS memory grows monotonically** — `torch.mps.current_allocated_memory()` increases with each call and never recedes naturally
2. **The degradation is in AR inference only** — vocoding (S3Gen/HiFiGAN) stays fast, suggesting the T3 autoregressive loop is the bottleneck
3. **`torch.mps.empty_cache()` between calls prevents the degradation** — adding this single call after each `synthesize()` keeps times consistent

## Workaround

Calling `torch.mps.empty_cache()` after each synthesis invocation prevents the issue:

```python
import torch

for text in sentences:
    async for chunk in tts.synthesize(text, voice="builtin"):
        process(chunk)
    
    # Prevent MPS memory accumulation
    if torch.backends.mps.is_available():
        torch.mps.empty_cache()
```

## Suggested Fix

ChatterStream should call `torch.mps.empty_cache()` internally after each `synthesize()` iteration completes when running on the MPS backend. This could be added at the end of the `synthesize()` async generator in `StreamingTTS`, or in the `TokenGenerator` after the AR loop finishes.

A pre-warm pass during `load()` followed by `torch.mps.empty_cache()` would also help establish a clean baseline for the MPS allocator before real requests begin.

### Additional context

The HLS streaming example (`examples/streaming_server.py`) in this repo does **not** exhibit this issue because:
1. It uses an `asyncio.Lock` to serialize synthesis (no thread pool)
2. Sessions are cleaned up after use
3. The pattern of streaming segments incrementally means intermediate tensors are released sooner

The issue is most pronounced when synthesis is called from a thread pool (`run_in_executor`) with a new event loop per call, which prevents MPS tensors from being garbage collected between invocations. Even without threading, the issue reproduces on the main event loop — it just takes more calls to become noticeable.
